docs/content/docs/recipes/shared-string.md

```diff
@@ -0,0 +1,454 @@
+    }
+    replaceText(text, start, end) {
+      this._sharedString.replaceText(start, end, text);
+    }
+    ```
+
+1. Replace `TODO 3` with the following code. Note about this code.
+    - This function handles changes made to the `_sharedString` object.
+    - `transformPosition` is a function which will give the new position of the caret given the oldPosition. This function must be defined based on the value of `op.type`.
+
+    ```js
+    sequenceDeltaHandler = (event) => {
+      this._latestText = this._sharedString.getText();
+      const isLocal = event.isLocal;
+
+      const op = event.opArgs.op;
+      let transformPosition;
+      if (op.type === MergeTreeDeltaType.INSERT) {
+        // TODO 3A: Define transformPosition for an INSERT operation
+      } else if (op.type === MergeTreeDeltaType.REMOVE) {
+        // TODO 3B: Define transformPosition for a REMOVE operation
+      } else {
+        throw new Error("Don't know how to handle op types beyond insert and remove");
+      }
+
+      this.emit("textChanged", { isLocal, transformPosition });
+    };
+    ```
+
+    Replace `TODO 3A` with the following code to handle an `INSERT` operation.
+
+    ```js
+    transformPosition = (oldPosition) => {
+      if (op.pos1 === undefined) {
+        throw new Error("pos1 undefined");
+      }
+      if (op.seg === undefined) {
+        throw new Error("seg undefined");
+      }
+      const changeStartPosition = op.pos1;
+      const changeLength = (op.seg).length;
+      let newPosition;
+      if (oldPosition <= changeStartPosition) {
+        // Position is unmoved by the insertion if it is before the insertion's start
+        newPosition = oldPosition;
+      } else {
+        // Position is moved by the length of the insertion if it is after the insertion's start
+        newPosition = oldPosition + changeLength;
+      }
+      return newPosition;
+    };
+    ```
+
+    Replace `TODO 3B` with the following code to handle a `REMOVE` operation.
+
+    ```js
+    transformPosition = (oldPosition) => {
+      if (op.pos1 === undefined) {
+        throw new Error("pos1 undefined");
+      }
+      if (op.pos2 === undefined) {
+        throw new Error("pos2 undefined");
+      }
+      const changeStartPosition = op.pos1;
+      const changeEndPosition = op.pos2;
+      const changeLength = changeEndPosition - changeStartPosition;
+      let newPosition;
+      if (oldPosition <= changeStartPosition) {
+        // Position is unmoved by the deletion if it is before the deletion's start
+        newPosition = oldPosition;
+      } else if (oldPosition > (changeEndPosition - 1)) {
+        // Position is moved by the size of the deletion if it is after the deletion's end
+        newPosition = oldPosition - changeLength;
+      } else {
+        // Position snaps to the left side of the deletion if it is inside the deletion.
+        newPosition = changeStartPosition;
+      }
+      return newPosition;
+    };
+    ```
```

Comment:
Felt it may be a good idea to break up TODO 3 into TODO 3A and 3B since otherwise it may be too much for a single code block

Reply:
seeing this here makes me think this might be out of scope for this example and would be better suited as an import.  like, i don't think this is too different from delving into SharedString internals, which is more clearly out of scope.

Reply:
I think this makes sense. Only concern may be using an import from `fluid-experimental`. @tylerbutler , would you mind giving your opinion?

Reply:
Yeah, experimental isn't great. That said, I don't want to block on that. Just go ahead and pull in the experimental import and file an issue to track making sure this function (or the whole package) is eventually moved out of experimental.